src/plugins/telemetry_collection_manager/server/types.ts

```diff
@@ -157,5 +162,6 @@ export interface Collection<
   clusterDetailsGetter: ClusterDetailsGetter<CustomContext>;
   esCluster: ILegacyClusterClient;
   esClientGetter: () => IClusterClient | undefined; // the collection could still return undefined for the es client getter.
+  soServiceGetter: () => SavedObjectsServiceStart | undefined; // the collection could still return undefined for the Saved Objects Service getter.
```

Comment:
irrelated to PR: in what cases it would return `undefined`?

Reply:
The saved objects client is only available during the start cycle, so if it's called before it's initialized, we'll return undefined. It's the same as for the new elasticsearch client.